Fix panic with new workspaces

luca-della-vedova opened the pull request:
## Bug fix

### Fixed bug

Fixes #130.

### Fix applied

The fix is reasonably straightforward, just change the `Default` implementation of sites to have a new empty level. I'm not 100% sure putting a site ID of 0 is sound, and whether this would create any issue if we had multiple workspaces open

mxgrey commented:
I don't think this is the way we should fix the issue. `rmf_site_format` is meant to be used by more than the editor GUI. It's also meant for use by scripts or other programs which may be surprised when a default initialization of a site gets automatically filled with exactly one level.

If anything, I think we should modify [this code](https://github.com/open-rmf/rmf_site/blob/f796e44ce5a1d13a2878b5cc2cd7d93de0bbee0b/rmf_site_editor/src/workspace.rs#L141) to create a default (empty) site and add one level before we dispatch it to `load_site`.

luca-della-vedova commented:
> I don't think this is the way we should fix the issue. `rmf_site_format` is meant to be used by more than the editor GUI. It's also meant for use by scripts or other programs which may be surprised when a default initialization of a site gets automatically filled with exactly one level.
> 
> If anything, I think we should modify [this code](https://github.com/open-rmf/rmf_site/blob/f796e44ce5a1d13a2878b5cc2cd7d93de0bbee0b/rmf_site_editor/src/workspace.rs#L141) to create a default (empty) site and add one level before we dispatch it to `load_site`.

Alright! Reverted the original fix and changed at the event sending level https://github.com/open-rmf/rmf_site/pull/131/commits/2be086317ffc06c95109c6784fa3523fc3462dbc
Added a `Default` derive for `Level` as well, all the fields implement default anyway so that helps reducing boilerplate

arjo129 reviewed (approved):
Works for me!
